src/monitoring_service/database.py

```diff
@@ -143,6 +157,94 @@ def get_channel(self, token_network_address: str, channel_id: int) -> Optional[C
     def channel_count(self) -> int:
         return self.conn.execute("SELECT count(*) FROM channel").fetchone()[0]
 
+    def upsert_scheduled_event(self, event: ScheduledEvent) -> None:
+        contained_event: SubEvent = cast(SubEvent, event.event)
+        values = [
+            hex(event.trigger_block_number),
+            EVENT_TYPE_ID_MAP[type(contained_event)],
+            contained_event.token_network_address,
+            hex(contained_event.channel_identifier),
+            contained_event.non_closing_participant,
+        ]
+        with self.conn:
+            upsert_sql = "INSERT OR REPLACE INTO scheduled_events VALUES ({})".format(
+                ', '.join('?' * len(values)),
+            )
+            self.conn.execute(upsert_sql, values)
+
+    def get_scheduled_events(
+            self,
+            max_trigger_block: int,
+    ) -> List[ScheduledEvent]:
+        rows = self.conn.execute(
+            """
+                SELECT * FROM scheduled_events
+                WHERE trigger_block_number <= ?
+            """,
+            [hex(max_trigger_block)],
+        ).fetchall()
+
+        def create_scheduled_event(row: sqlite3.Row) -> ScheduledEvent:
+            klass = EVENT_ID_TYPE_MAP[row['event_type']]
```

Comment:
The variable could be name `EventType`, which is more descriptive and gets around the spelling-mistake-to-avoid-keyword uglyness.